File: benchmarks/raytracer/raytracer-assemblyscript/assembly/sphere.ts
import { Vector } from './vector';
import { Material } from './material';

/*
 * Class holding data about a sphere
 */
export class Sphere {

	// Dummy sphere
	static NONE: Sphere = new Sphere(Vector.ZERO, 0, Material.NONE);

	constructor(
		// the position of the center of this sphere 
		public center: Vector,
		// the radius of this sphere
		public radius: f32,
		// the surface material
		public material: Material,
	) {}


}

File: benchmarks/raytracer/raytracer-rust/src/sphere.rs
use crate::vector::Vector;
use crate::material::Material;

/*
 * Struct holding data about a sphere 
 */
 #[derive(Copy, Clone)]
pub struct Sphere {
    // the position of the center of this sphere 
    pub center: Vector,
    // the radius of this sphere
    pub radius: f32,
    // the surface material
    pub material: Material,
}

impl Sphere {
    // dummy sphere
    pub const NONE: Sphere = Sphere {
        center: Vector::ZERO,
        radius: 0.0,
        material: Material::NONE
    };
}